libs/books/data-access/src/lib/+state/reading-list.effects.ts

```diff
@@ -54,9 +61,56 @@ export class ReadingListEffects implements OnInitEffects {
     )
   );
 
+  undoAddBook$ = createEffect(() =>
```

Comment:
reconsider this logic as you are calling the http calls for add/remove again for undo effects, try to reuse the add/remove effects